src/bitmovin-dotnet/Resource/HlsResource.cs

```diff
@@ -9,6 +10,40 @@ public HlsResource(RestClient client, string url) : base(client, url)
         {
         }
 
+#if !NET_40
+
+        public async Task<StreamInfo> AddStreamInfoAsync(string id, StreamInfo s)
+        {
+            var postUrl = string.Format("{0}/{1}/streams", _url, id);
+            return await _restClient.PostAsync<StreamInfo>(postUrl, s);
+        }
+
+        public async Task<MediaInfo> AddMediaInfoAsync(string id, MediaInfo s)
+        {
+            var postUrl = string.Format("{0}/{1}/media", _url, id);
+            return await _restClient.PostAsync<MediaInfo>(postUrl, s);
+        }
+
+        public async Task<string> StartAsync(string id)
+        {
+            var postUrl = string.Format("{0}/{1}/start", _url, id);
+            return await _restClient.PostAndGetIdAsync(postUrl);
+        }
+
+        public async Task<string> StopAsync(string id)
+        {
+            var postUrl = string.Format("{0}/{1}/stop", _url, id);
+            return await _restClient.PostAndGetIdAsync(postUrl);
+        }
+
+        public async Task<Task> RetrieveStatusAsync(string id)
```

Comment:
This signature looks a bit weird.. What is what? Threading.Task or Rest.Task?
I'd change the using on top to a EncodingTask or something.. It's very very confusing otherwise.